### ISSUE TITLE
Report grouping 

### ISSUE THREAD

**Issue by manfredu** [mtierltd/timetracker#142]
Timetracker is a very useful application, thank you for providing and maintaining it!  

I'm using 0.0.69 and for me report grouping does not work as expected. E.g., I want to see daily sums, but still get several lines. Also for  weekly sums I sometimes get several entries. Monthly and yearly sums seem to be fine.

Additional question: What is shown in the name colounm? Description ist displayed as * when grouping, shouldn't that happen for name as well?

**Comment by roynico** [comment]
Hej! I encounter a similar problem. "reports", "group by daily" -> It displays "ERROR" and then an empty table.

**Comment by smatthie** [comment]
Same problem here, in particular the "group by: daily" never does any grouping by day. Depending on the other settings it either gives a list of all entries or it displays "ERROR" (and then I have to reload the whole page to recover).

**Comment by mtierltd** [comment]
Hi

Please test commit 44ff8392da4be24f073c7ede0f8711f9527424ef

**Comment by smatthie** [comment]
@mtierltd is that commit in version 0.0.72? It's still not working in that version.

**Comment by mtierltd** [comment]
Yes. We have done tests with both MariaDB and PostgreSQL and we could not reproduce the issue anymore with 0.0.72.

**Comment by smatthie** [comment]
Hmm, still doesn't work for me. The "daily" option always just returns a list of all individual items, no matter what I chose in the other two fields. For the weekly and monthly I do get (as expected) a single table row for each week/month that sums up all the entries. I'll try to do some more tests and see what exactly is going on, but that may take a while.

**Comment by smatthie** [comment]
I had a look at `lib/Db/ReportItemMapper.php` and there's something I don't understand. In lines 82-90 you have conditional statements that test for week, year, month but not actually for day, so it looks to me that when you select the daily grouping it leads to the last ELSE which doesn't group at all.

**Comment by smatthie** [comment]
So I had a look at `lib/Db/ReportItemMapper.php` and I can't really work out where the grouping by day happens. For example, in lines 82-90 you have conditional statements for week, year and month, but not for day, so the day grouping does the same as the none grouping, i.e. no grouping at all?

**Comment by smatthie** [comment]
It works when I add between lines 87 and 88:
`}elseif ($timegroup == 'day') {
                $selectFields[]= "DATE(start) as ftime";`

**Comment by smatthie** [comment]
... there's also a time zone problem that the date or week is shifted by one if I was working between 23:00 and midnight, but that's a separate thing.

**Comment by manfredu** [comment]
For me report grouping is working fine now! Thanks!

A small remaining issue is column "Name" which sometimes contains a random value. E.g. when group by "Daily" the sum of the day is correctly displayed as one entry.

**Comment by manfredu** [comment]
When group by is set to "weekly" or "monthly" sometimes I get more than one entry. Looks like this happens when the content of column name is different. So this seems to be inconsistent with "daily".

**Comment by manfredu** [comment]
Or maybe the duplicate entries occur when working over midnight and the time entry belongs to more than one week or month...?
